extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -0,0 +1,167 @@
+		this.optionsModel = new DeployOptionsModel(defaultOptions);
+	}
+
+	protected initializeDialog(): void {
+		this.optionsTab = utils.getAzdataApi()!.window.createTab(constants.GeneralOptions);
+		this.intializeDeploymentOptionsDialogTab();
+		this.dialog.content = [this.optionsTab];
+	}
+
+	public openDialog(): void {
+		this.dialog = utils.getAzdataApi()!.window.createModelViewDialog(constants.GeneralOptions);
+
+		this.initializeDialog();
+
+		this.dialog.okButton.label = constants.okString;
+		this.dialog.okButton.onClick(async () => this.execute());
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+		this.dialog.cancelButton.onClick(async () => this.cancel());
+
+		let resetButton = utils.getAzdataApi()!.window.createButton(constants.ResetButton);
+		resetButton.onClick(async () => await this.reset());
+		this.dialog.customButtons = [];
+		this.dialog.customButtons.push(resetButton);
+
+		utils.getAzdataApi()!.window.openDialog(this.dialog);
+	}
+
+	private intializeDeploymentOptionsDialogTab(): void {
+		this.optionsTab?.registerContent(async view => {
+			this.descriptionHeading = view.modelBuilder.table().withProps({
+				data: [],
+				columns: [
+					{
+						value: constants.OptionDescription,
+						headerCssClass: 'no-borders',
+						toolTip: constants.OptionDescription
+					}
+				]
+			}).component();
+
+			this.descriptionText = view.modelBuilder.text().withProps({
+				value: ' '
+			}).component();
+
+			this.optionsTable = view.modelBuilder.table().component();
+			await this.updateOptionsTable();
+
+			// Get the description of the selected option
+			this.disposableListeners.push(this.optionsTable.onRowSelected(async () => {
+				const row = this.optionsTable.selectedRows![0];
+				let label = this.optionsModel.optionsLabels[row];
+				await this.descriptionText.updateProperties({
+					value: this.optionsModel.getDescription(label)
+				});
+			}));
+
+			// Update deploy options value on checkbox onchange
+			this.disposableListeners.push(this.optionsTable.onCellAction!((rowState) => {
+				let checkboxState = <azdataType.ICheckboxCellActionEventArgs>rowState;
+				if (checkboxState && checkboxState.row !== undefined) {
+					let label = this.optionsModel.optionsLabels[checkboxState.row];
+					this.optionsModel.optionsLookup[label] = checkboxState.checked;
+				}
+			}));
+
+			this.optionsFlexBuilder = view.modelBuilder.flexContainer()
+				.withLayout({
+					flexFlow: 'column'
+				}).component();
+
+			this.optionsFlexBuilder.addItem(this.optionsTable, { CSSStyles: { 'overflow': 'scroll', 'height': '65vh' } });
+			this.optionsFlexBuilder.addItem(this.descriptionHeading, { CSSStyles: { 'font-weight': 'bold', 'height': '30px' } });
+			this.optionsFlexBuilder.addItem(this.descriptionText, { CSSStyles: { 'padding': '4px', 'margin-right': '10px', 'overflow': 'scroll', 'height': '10vh' } });
+			await view.initializeModel(this.optionsFlexBuilder);
+			await this.optionsTable.focus();
```

Comment:
does the focus go to the first checkbox when the dialog is opened?

Reply:
yes, this will help to display the focused(first option)'s description.